.github/workflows/build.yml

```diff
@@ -192,6 +192,76 @@ jobs:
         run: cargo test --locked --release --no-run
         working-directory: rustls
 
+  overwrite-atomic-sync-build-targets:
```

Comment:
I don't think we should merge this -- I don't think this repository's maintainers should be on the hook for enforcing that this works, given the level of complexity here.

Reply:
Ah! To add to the malaise I'm not sure I support this PR without some sort of defense of the changes.

I wonder if clippy `disallowed-types` (and co) could be used instead? (With a specific `allow` just for `crate::atomic_sync`?

Reply:
This is now gone, I have updated the clippy configuration as suggested. I think using this clippy configuration together with the `overwrite-atomic-sync-test` job should be good enough to validate this. A downstream fork can always do quick hacks & adaptations if needed while waiting for potential future regressions.

Reply:
I'd still prefer to skip all the added CI stuff except for the clippy disallowed types bits. I think the latter is sufficient to help us keep this working, and I'd rather avoid us maintaining all the cargo/bash stuff that patch rustls in CI.

Reply:
I can remove this but thought there was some interest in improving some of this CI stuff from these comments like this: https://github.com/rustls/rustls/pull/2285#discussion_r1923792155

I did some work on improving this in response to other comments.

This also would have been a nice demonstration, as pointed out by verbiage suggested by @cpu: https://github.com/rustls/rustls/pull/2285#discussion_r1923812261

Reply:
> I can remove this but thought there was some interest in improving some of this CI stuff from these comments like this: https://github.com/rustls/rustls/pull/2285#discussion_r1923792155

I made those comments in the spirit of improving the implementation if consensus was that it was the right overall direction. I've personally been swayed by djc's argument that it makes the most sense to not have it in this repo. I think @ctz is also supportive of that direction (?)

Sorry if that made some extra work.

Reply:
I will remove the CI additions then. We can always add this in a separate PR, if (ever) desired. Yes "extra work" is sometimes normal part of the game, sometimes needed till we get a chance to make the final decision.

Reply:
I have removed the CI additions and rebased. Please let me know if anything else is needed from me.

The CI additions would have been nice but unfortunately with some cost of additional complexity and potentially a little confusing for newcomers. Maybe a separate CI yml script with some better structure & more comments. We can always add this later if someone has extra time to look into this.

This would totally help another fork support these targets with minimal adjustments needed. I may do this with a few more things as well, supporting the idea in #2137 with `portable-io` for example.